message_center_compassion/mappings/base_mapping.py

```diff
@@ -287,13 +297,16 @@ def itersubclasses(cls, _seen=None):
                 yield sub
 
 
-def new_onramp_mapping(model, env):
+def new_onramp_mapping(model, env, action='default'):
     """Return an instance of the good Mapping class based on the given model.
 
     :param model: model name of the mapping.
+    :param action: action name (not mandatory, specify when a model has many
+    actions)
     :return: class instance for given model mapping.
     """
     for cls in itersubclasses(OnrampMapping):
         if cls.ODOO_MODEL == model:
-            return cls(env)
+            if cls.ACTION == action:

```

Comment:
could be done in the same test `if cls.ODOO_MODEL == model and cls.ACTION == action`